src/main/java/io/strimzi/test/container/StrimziZookeeperContainer.java

```diff
@@ -24,35 +24,52 @@
-
+    private static LogicalKafkaVersionEntity logicalKafkaVersionEntity;
     private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
 
-    /**
-     * Default ZooKeeper port
-     */
-    public static final int ZOOKEEPER_PORT = 2181;
+    static {
+        logicalKafkaVersionEntity = new LogicalKafkaVersionEntity();
+    }
 
-    private int zookeeperExposedPort;
+    private StrimziZookeeperContainer(StrimziZookeeperContainerBuilder builder) {
+        if (builder.strimziTestContainerVersion == null || builder.strimziTestContainerVersion.isEmpty()) {
+            this.strimziTestContainerVersion = logicalKafkaVersionEntity.latestRelease().getStrimziTestContainerVersion();
+            LOGGER.info("You did not specify Strimzi test container version. Using latest release:{}", this.strimziTestContainerVersion);
+        } else {
+            this.strimziTestContainerVersion = builder.strimziTestContainerVersion;
+        }
+        if (builder.kafkaVersion == null || builder.kafkaVersion.isEmpty()) {
+            this.kafkaVersion = logicalKafkaVersionEntity.latestRelease().getVersion();
+            LOGGER.info("You did not specify Kafka version. Using latest release:{}", this.kafkaVersion);
+        } else {
+            this.kafkaVersion = builder.kafkaVersion;
+        }
+
+        this.setDockerImageName("quay.io/strimzi-test-container/test-container:" +
+            this.strimziTestContainerVersion + "-kafka-" +
+            this.kafkaVersion);
 
-    /**
-     * Constructor for @code{StrimziZookeeperContainer}, which instantiate object based on env variables.
-     */
-    public StrimziZookeeperContainer() {
-        super("quay.io/strimzi-test-container/test-container:" +
-            Environment.getValue(Environment.STRIMZI_TEST_CONTAINER_IMAGE_VERSION_ENV) + "-kafka-" +
-            Environment.getValue(Environment.STRIMZI_TEST_CONTAINER_KAFKA_VERSION_ENV));
         // we need this shared network in case we deploy StrimziKafkaCluster, which consist `StrimziZookeeperContainer`
         // instance and by default each container has its own network
-        super.withNetwork(Network.SHARED);
+        this.withNetwork(Network.SHARED);
         // exposing zookeeper port from the container
-        withExposedPorts(ZOOKEEPER_PORT);
-        withNetworkAliases("zookeeper");
-        withEnv("LOG_DIR", "/tmp");
-        withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(ZOOKEEPER_PORT));
+        this.withExposedPorts(ZOOKEEPER_PORT);
+        this.withNetworkAliases("zookeeper");
+        this.withEnv("LOG_DIR", "/tmp");
+        this.withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(ZOOKEEPER_PORT));
         // env for readiness
-        withEnv("ZOO_4LW_COMMANDS_WHITELIST", "ruok");
+        this.withEnv("ZOO_4LW_COMMANDS_WHITELIST", "ruok");
     }
 
+    /**
+     * Default ZooKeeper port
+     */
+    public static final int ZOOKEEPER_PORT = 2181;
```

Comment:
Sae as in the other class.

Reply:
I have created a `Constants` for this one... :)

Reply:
It was more the ordering of class fields, methods etc. I meant here really - not the existence of this field in this class (I don't necessarily mind having a Constants class ... but that wasn't really how I meant it here.)